maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/copyright/ValidatorBase.java

```diff
@@ -67,8 +67,14 @@ public void validate(FileRequest file, Path path) {
 
         try (BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_8)) {
             copyrightComment = readComment(file, br);
-        } catch (IOException e) {
-            throw new EnforcerException("Failed to read file " + path, e);
+        } catch (IOException e1) {
+            //Fallback to cover at least UTF-16 encoding
+            //Encoding autodetection should be better, but depends on 3rd party library
+            try (BufferedReader br = Files.newBufferedReader(path, StandardCharsets.UTF_16)) {
```

Comment:
Can you fallback on UTF-16 only if e1 is an instance of `CharacterCodingException` ?

Reply:
OK, changed, but `IOException` has to be handled too due `Files.newBufferedReader` and `readComment(`.